packages/mui-material/src/styles/ThemeProviderNoVars.tsx

```diff
@@ -8,11 +8,25 @@ export interface ThemeProviderNoVarsProps<Theme = DefaultTheme> {
   theme: Partial<Theme> | ((outerTheme: Theme) => Theme);
 }
 
+let warnedOnceNoSsr = false;
+
 export default function ThemeProviderNoVars<Theme = DefaultTheme>({
   theme: themeInput,
   ...props
 }: ThemeProviderNoVarsProps<Theme>): React.ReactElement<ThemeProviderNoVarsProps<Theme>> {
   const scopedTheme = THEME_ID in themeInput ? themeInput[THEME_ID] : undefined;
+  if (process.env.NODE_ENV !== 'production') {
+    // @ts-ignore
+    if (process.env.NODE_ENV !== 'test' && props.noSsr && !warnedOnceNoSsr) {
+      console.error(
+        [
+          'MUI: The `noSsr` prop must be used with the theme that contains light and dark colorSchemes.',
+          'See https://mui.com/material-ui/customization/dark-mode/#built-in-support for more details.',
```

Comment:
Same feedback as in https://github.com/mui/material-ui/pull/43783#discussion_r1844264494.